extensions/iceberg/src/main/java/io/deephaven/iceberg/util/IcebergTableWriter.java

```diff
@@ -129,6 +144,21 @@ private static TableParquetWriterOptions verifyWriterOptions(
         return (TableParquetWriterOptions) tableWriterOptions;
     }
 
+    /**
+     * Return a {@link TableDefinition} which contains only the non-partitioning columns from the provided table
+     * definition.
+     */
+    private static List<ColumnDefinition<?>> nonPartitioningColumnDefinitions(
```

Comment:
we should just return `new TableDefinition(nonPartitioningColumns)`, in that way we could take advantage of `table.checkMutualCompatibility` (it automatically throws a nicely formatted error message). mutual compatibility is _almost_ the same as equals, but allows the columns to be in different orders. (We could advocate for a stricter `table.checkEquals`, but I'm not too worried about being completely strict like that.)